app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -95,7 +95,7 @@ class WifiDirectManager @Inject constructor(
     this.filesForTransfer = filesForTransfer
     isFileSender = filesForTransfer.isNotEmpty()
     manager = activity.getSystemService(Context.WIFI_P2P_SERVICE) as WifiP2pManager
-    channel = manager.initialize(activity, Looper.getMainLooper(), null)
+    channel = manager.initialize(activity.baseContext, Looper.getMainLooper(), null)
```

Comment:
This makes me question if we need an activity at all. Change the type of Activity to Context and try to transfer a file

Reply:
>try to transfer a file

It works fine.
>Change the type of Activity to Context

It is already a type of context.

Reply:
When you inject a `Context` our Dagger setup is such that it returns the application context, baseContext opens up the hazy realm of ContextWrappers that are best to avoid

Reply:
>This makes me question if we need an activity at all. Change the type of Activity to Context and try to transfer a file
it works fine.

Reply:
groovy